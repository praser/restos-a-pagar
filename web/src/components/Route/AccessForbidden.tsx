import React from 'react';
import Error from 'pages/Error';

const AccessForbidden = () => (
  <Error
    code={401}
    description="Você não pode acessar essa página"
    paragraph="Há uma diferença entre conhecer o caminho e percorrer o caminho."
  />
);

export default AccessForbidden;
