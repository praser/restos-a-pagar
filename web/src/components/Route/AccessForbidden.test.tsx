import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import AccessForbidden from 'components/Route/AccessForbidden';

describe('AccessForbiden Component', () => {
  it('is expected to render correct component', () => {
    const component = renderer.create(<AccessForbidden />).toJSON();
    expect(component).toMatchSnapshot();
  });

  describe('screen', () => {
    beforeEach(() => {
      render(<AccessForbidden />);
    });

    it('is expected to show the http code', () => {
      const text = '401';
      expect(screen.getByText(text)).toBeInTheDocument();
    });

    it('expected to show the descript', () => {
      const text = 'Você não pode acessar essa página';
      expect(screen.getByText(text)).toBeInTheDocument();
    });

    it('is expect to show a long descriptio', () => {
      const text =
        'Há uma diferença entre conhecer o caminho e percorrer o caminho.';
      expect(screen.getByText(text)).toBeInTheDocument();
    });
  });
});
