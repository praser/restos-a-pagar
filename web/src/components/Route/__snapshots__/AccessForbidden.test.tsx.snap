// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccessForbiden Component is expected to render correct component 1`] = `
<div
  className="sc-kEjbxe dhrXyX"
>
  <nav
    className="sc-bdfBwQ fIcMTq"
  >
    <h1
      className="sc-gsTCUz kfVslw"
    >
      Painel de gestão do RAP
    </h1>
    <ul
      className="sc-dlfnbm fhAIwl"
    >
      <div
        className="sc-hKgILt hSuYkk"
      />
      <li
        className="sc-gKsewC gMZBwq"
      >
        <button
          className="sc-iBPRYJ fCMuWI"
          onClick={[Function]}
        >
          <span
            className="sc-fubCfw fDxNAs"
          />
          <img
            className="sc-pFZIQ kCLbWH"
            src="undraw_male_avatar_323b.svg"
          />
        </button>
        <div
          className="sc-eCssSg jacAYx"
        >
          <a
            className="sc-jSgupP iHQRUU"
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-sign-in-alt fa-w-16 sc-jrAGrp fzTpjZ"
              data-icon="sign-in-alt"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 448h-84c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h84c17.7 0 32-14.3 32-32V160c0-17.7-14.3-32-32-32h-84c-6.6 0-12-5.4-12-12V76c0-6.6 5.4-12 12-12h84c53 0 96 43 96 96v192c0 53-43 96-96 96zm-47-201L201 79c-15-15-41-4.5-41 17v96H24c-13.3 0-24 10.7-24 24v96c0 13.3 10.7 24 24 24h136v96c0 21.5 26 32 41 17l168-168c9.3-9.4 9.3-24.6 0-34z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            Sair
          </a>
        </div>
      </li>
    </ul>
  </nav>
  <div
    className="sc-iqHYGH jKTyr"
  >
    <div
      className="sc-crrsfI fEyrVP"
    >
      401
    </div>
    <p
      className="sc-dQppl kFVvED"
    >
      Você não pode acessar essa página
    </p>
    <p
      className="sc-bqyKva ekScID"
    >
      Há uma diferença entre conhecer o caminho e percorrer o caminho.
    </p>
    <button
      className="sc-kstrdz clWmeK"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-long-arrow-alt-left fa-w-14 "
        data-icon="long-arrow-alt-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M134.059 296H436c6.627 0 12-5.373 12-12v-56c0-6.627-5.373-12-12-12H134.059v-46.059c0-21.382-25.851-32.09-40.971-16.971L7.029 239.029c-9.373 9.373-9.373 24.569 0 33.941l86.059 86.059c15.119 15.119 40.971 4.411 40.971-16.971V296z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
       Voltar para a página anterior
    </button>
  </div>
</div>
`;
